### PR TITLE
Configured GitHub social media authentication using Firebase

### LINKED ISSUE #9: Splash Page
A splash page that includes navbar, a centered title element, and social media login buttons.

### LINKED ISSUE #10: Firebase Authentication
Setup Firebase to allow authentication over social media with project.

### PR DESCRIPTION
**Pull request by adelhelj**
Configured GitHub Auth
To get started, cd into the frontend directory
Run 'npm install' and 'npm install firebase' 
Then type 'npm start'
The firebaseConfig.js file credentials are not on this, so I can share with you. I'm also working on .env variable token.

Apple auth required an Apple dev subscription, Meta required a privacy policy site, Twitter required a hosting site (no localhost:3000). Perhaps we can implement those down the line, but for now the Google and GitHub user sign up features work. 

Closes #9 and closes #10

### CODE REVIEW

File: frontend/src/components/SplashPage.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import React from 'react';
+
 import '../styles/SplashPage.css';
-import { getAuth, signInWithPopup, GoogleAuthProvider } from 'firebase/auth';
+import { getAuth, signInWithPopup, GoogleAuthProvider, GithubAuthProvider, createUserWithEmailAndPassword } from 'firebase/auth';
```
Comment (GPropersi) [inline frontend/src/components/SplashPage.js:4]: Are we going to allow creation with email and password? Wasn't sure we could handle that

**Review by GPropersi — COMMENTED** [review]
Looks good to me!

**Review by barreryu — COMMENTED** [review]
Good job! I see that works on your end based off the screenshots, but I'm having issues on my end with timeout errors.

**Review by barreryu — APPROVED** [review]
Sorry never mind, I was able to run the authentication now! Good work!

File: frontend/src/components/SplashPage.js (hunk 1 above)
Comment (adelhelj) [reply frontend/src/components/SplashPage.js:4]: We can remove that imported class. Was just experimenting with forms when I saw that Facebook, Twitter, and Microsoft authentication was a bit more complex than Google, GitHub. Originally wanted 3 creation techniques prior to our Friday meeting last week.

**Review by GPropersi — APPROVED** [review]

**Pull request merged**